### PR TITLE
Testing dom-to-image functionality

### PR DESCRIPTION
**Pull request by idovcohen**

### CODE REVIEW

File: .angulardoc.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "repoId": "572b59cf-0810-4343-9d09-05607f5003bd",
+  "lastSync": 0
+}
```
Comment (asafMasa) [inline .angulardoc.json]: what is this file?
add enter at the end

File: src/app/@ansyn/imagery/model/base-imagery-map.ts [hunk 2]
```diff
@@ -40,6 +40,10 @@ export abstract class BaseImageryMap<T = any> {
 
 	abstract getCenter(): Observable<Point>;
 
+	public getTargetElement(): HTMLElement {
```
Comment (asafMasa) [inline src/app/@ansyn/imagery/model/base-imagery-map.ts]: why not an abstract method?

File: src/app/@ansyn/map-facade/services/map-facade.service.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
+import { CommunicatorEntity } from './../../imagery/communicator-service/communicator.entity';
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/services/map-facade.service.ts]: use @ansyn/imagery

File: src/app/@ansyn/map-facade/services/map-facade.service.ts [hunk 4]
```diff
@@ -99,26 +101,28 @@ export class MapFacadeService {
...
-				this.store.dispatch(new ExportMapsToPngActionSuccess());
-			}),
-			catchError((err) => {
-				this.store.dispatch(new ExportMapsToPngActionFailed(err));
-				return EMPTY;
-			})
-		).subscribe();
+	async exportMapsToPng(element: Element | string = null) {
+		this.store.dispatch(new SetMinimalistViewModeAction(true));
+		await setTimeout(async () => {
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/services/map-facade.service.ts]: i think you should remove this method from here
add action of start export, end export, and in the map facade register to it and do your thing there

File: src/app/@ansyn/imagery/model/base-imagery-map.ts (hunk 2 above)
Comment (idovcohen) [reply src/app/@ansyn/imagery/model/base-imagery-map.ts]: This way you don't need to implement it evetywhere.
Actually... I will remove it. don't need it anymore.

File: src/app/@ansyn/map-facade/reducers/map.reducer.ts [hunk 5]
```diff
@@ -194,6 +196,15 @@ export function MapReducer(state: IMapState = initialMapState, action: MapAction
 		case MapActionTypes.SET_MINIMALIST_VIEW_MODE:
 			return { ...state, minimalistViewMode: action.payload };
 
+		case MapActionTypes.EXPORT_MAPS_TO_PNG_REQUESTED:
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/reducers/map.reducer.ts]: add to logs

File: src/app/@ansyn/map-facade/components/imageries-manager/imageries-manager.component.ts [hunk 6]
```diff
@@ -136,10 +145,62 @@ export class ImageriesManagerComponent implements OnInit, AfterContentChecked {
...
+	private async exportElement(element: Element): Promise<{result: boolean, msg: string}> {
+		try {
+			const blob: Blob = await domtoimage.toBlob(element,
+								{filter: (node) => {
+									return (node.id !== 'loader'); }});
+			saveAs(blob, 'map.jpeg');
+			return {result: true, msg: null};
+		}
+		catch (err) {
+			console.error(err);
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/components/imageries-manager/imageries-manager.component.ts]: remove

Comment (asafMasa) [reply src/app/@ansyn/map-facade/components/imageries-manager/imageries-manager.component.ts]: 😃

**Pull request merged**